### ISSUE TITLE
CascadeForestClassifier for cross validation in sklearn

### ISSUE THREAD

**Issue by pjgao** [LAMDA-NJU/Deep-Forest#2]
`CascadeForestClassifier` class cannot be used in sklearn's `cross_val_score` directly,  maybe we can inherited from `BaseEstimator`?
Something like this:
```python
from deepforest import CascadeForestClassifier
from sklearn.base import BaseEstimator
from sklearn.datasets import load_breast_cancer
from sklearn.metrics import accuracy_score
from sklearn.model_selection import cross_val_score

X, y = load_breast_cancer(return_X_y=True)


class CFC(CascadeForestClassifier, BaseEstimator):
    def __init__(self, **kwargs):
        super().__init__(**kwargs)

    def score(self, X, y):
        return accuracy_score(y, self.predict(X))


score = cross_val_score(CFC(random_state=10), X, y)
print(score)
```

**Comment by xuyxu** [comment]
You are right ;-). I am not aware of  the ``cross_val_score`` during the initial development.

I would appreciate a pull request very much, feel free to tell me if you need any help!

**Comment by xuyxu** [comment]
Closed via PR #8
